### ISSUE TITLE
add github CI?

### ISSUE THREAD

**Issue by Kreijstal** [arpadbuermen/OpenVAF#1]
So that we can do CI on github

**Comment by arpadbuermen** [comment]
I have no idea how that is done. Would appreciate help from you. I know Pascal had CI. Do I need some service outside github? Is it free? 

I think a working CI would make sense when you finish porting to llvm18 and I merge your changes. 

A great test is Dietmar's library of models. If it builds without an error that is a good sign. 

I also have some circuit analyses in VACASK with corresponding CTest system tests. Those should uncover a large part of semantic errors in the compiler because VACASK uses almost all of OSDI.

**Comment by Kreijstal** [comment]
yeah I am trying to make cargo test work but I am uncovering some segfaults, it could be the way I ported it to llvm18

![image](https://github.com/user-attachments/assets/78f3aced-720a-40fc-b903-5fa4cbee2d7b)

**Comment by Kreijstal** [comment]
okay just discovered for tests you need to do `RUN_SLOW_TESTS=1 cargo test --release` although I am having some bugs on release

**Comment by Kreijstal** [comment]
some things are really beyond me, what I'll do is I'll make a pull with what I have + CI and wait till someone+time fixes it

**Comment by arpadbuermen** [comment]
Great work.

**Comment by arpadbuermen** [comment]
CI now finally works (in llvm18 branch). When it gets merged, the master branch will have it too.
